diff --git a/dist/index.cjs b/dist/index.cjs
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -975,6 +975,9 @@ var globalOptions = {
   disabledTypes: []
 };
 function imageSize(input) {
+  if (typeof input === "string") {
+    input = require("node:fs").readFileSync(input);
+  }
   const type = detector(input);
   if (typeof type !== "undefined") {
     if (globalOptions.disabledTypes.indexOf(type) > -1) {
diff --git a/dist/index.mjs b/dist/index.mjs
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -1,3 +1,4 @@
+import { readFileSync } from "node:fs";
 var decoder = new TextDecoder();
 var toUTF8String = (input, start = 0, end = input.length) => decoder.decode(input.slice(start, end));
 var toHexString = (input, start = 0, end = input.length) => input.slice(start, end).reduce((memo, i) => memo + `0${i.toString(16)}`.slice(-2), "");
@@ -971,6 +972,9 @@ var globalOptions = {
   disabledTypes: []
 };
 function imageSize(input) {
+  if (typeof input === "string") {
+    input = readFileSync(input);
+  }
   const type = detector(input);
   if (typeof type !== "undefined") {
     if (globalOptions.disabledTypes.indexOf(type) > -1) {